genesis/engine/solvers/rigid/array_class.py

```diff
@@ -4,14 +4,248 @@
 import taichi as ti
 
 import genesis as gs
+import numpy as np
 
 # we will use struct for DofsState and DofsInfo after Hugh adds array_struct feature to taichi
 DofsState = ti.template()
 DofsInfo = ti.template()
+GeomsState = ti.template()
+GeomsInfo = ti.template()
+GeomsInitAABB = ti.template()
+LinksState = ti.template()
+LinksInfo = ti.template()
+VertsInfo = ti.template()
+EdgesInfo = ti.template()
 
 
 @ti.data_oriented
 class RigidGlobalInfo:
-    def __init__(self, n_dofs: int, n_entities: int, n_geoms: int, f_batch: Callable):
+    def __init__(self, n_dofs: int, n_entities: int, n_geoms: int, _B: int, f_batch: Callable):
         self.n_awake_dofs = ti.field(dtype=gs.ti_int, shape=f_batch())
         self.awake_dofs = ti.field(dtype=gs.ti_int, shape=f_batch(n_dofs))
+
+        self.n_geoms = ti.field(dtype=gs.ti_int, shape=())
+        self.n_geoms[None] = n_geoms
```

Comment:
we don't need to store n_geoms, _B, etc in a variable. They could be accessed by (e.g. geoms_state.shape[0], geoms_state.shape[1])